lib/amo-client.js

```diff
@@ -151,6 +148,76 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
 };
 
 
+Client.prototype.downloadSignedFiles = function(signedFiles, options) {
+  options = _.assign({
+    createWriteStream: fs.createWriteStream,
+    request: this._request,
+    stdout: process.stdout,
+  }, options);
+  var self = this;
+  var allDownloads = [];
+  // Set some defaults so it looks like 0% before starting the download.
+  var dataExpected = 100;
+  var dataReceived = 0.1;
+
+  function showProgress() {
+    var amount = ((dataReceived / dataExpected) * 100).toFixed();
+    // Pad the percentage amount so that the line length is consistent.
+    // This should do something like '  0%', ' 25%', '100%'
+    var padding = Array(4 - amount.length).join(' ');
+    options.stdout.write('\r' +
+        'Downloading signed file(s): ' + padding + amount + '% ');
+  }
+
+  function download(fileUrl) {
+    return when.promise(function(resolve, reject) {
+      // The API will give us a signed file named in a sane way.
+      var fileName = path.join(process.cwd(),
+                               path.basename(url.parse(fileUrl).path));
+      var out = options.createWriteStream(fileName);
+
+      var req = options.request({
+        method: 'GET',
+        uri: fileUrl,
+      });
+
+      req.on('error', function (error) {
+        reject(error);
+      });
+
+      req.pipe(out);
+
+      req.on('response', function (data) {
+        dataExpected += parseInt(data.headers['content-length']);
+      });
+
+      req.on('data', function (chunk) {
+        dataReceived += chunk.length;
+        showProgress();
+      });
+
+      req.on('end', function() {
+        options.stdout.write('\n');  // end the progress output
+        resolve(fileName);
+      });
+    });
+  }
+
+  showProgress();
+  signedFiles.forEach(function(file) {
+    allDownloads.push(download(file.download_url));
+  });
+
+  return when.all(allDownloads).then(function(downloadedFiles) {
+    logger.log('Your signed add-on awaits:');
+    downloadedFiles.forEach(function(fileName) {
+      logger.log('    ' + fileName.replace(process.cwd(), '.'));
```

Comment:
The replace doesn't look super safe but it's probably fine. Is there some relative path helper you could use instead?